### PR TITLE
Add documentation for image-builder github action

### PR DESCRIPTION
**Pull request by KacperMalachowski**
**Description**

Changes proposed in this pull request:

- Add readme to github action
- Update diagrams in description of github to ado integration

**Related issue(s)**

See #9978

### CODE REVIEW

**Review by Sawthis — CHANGES_REQUESTED** [review]

File: .github/actions/image-builder/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+## **adoResult**
+
+Result status of the Azure DevOps execution
+
+## **images**
+
+JSON formatted array containing all images build during process.
+
+# How It Work?
```
Comment (Sawthis) [inline .github/actions/image-builder/README.md:87]:
```suggestion
# How It Works?
```

File: .github/actions/image-builder/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+Github action is using **europe-docker.pkg.dev/kyma-project/prod/image-builder** docker image to trigger pipeline in Azure DevOps (ADO). It's passing parameters using REST API provided by ADO and kaniko-build-config from `main` branch of `test-infra` repository.
```
Comment (Sawthis) [inline .github/actions/image-builder/README.md:89]:
```suggestion
Github action is using **europe-docker.pkg.dev/kyma-project/prod/image-builder** docker image to trigger pipeline in Azure DevOps (ADO). It passes parameters using REST API provided by ADO and kaniko-build-config from `main` branch of `test-infra` repository.
```

File: .github/actions/image-builder/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+During the ADO pipeline execution is checking for status to be reported, if execution ended it fetches status and logs.
```
Comment (Sawthis) [inline .github/actions/image-builder/README.md:91]:
```suggestion
During the ADO pipeline execution this action is checking for status to be reported, if execution ended it fetches status and logs.
```

File: .github/actions/image-builder/README.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+If execution failed, the github action is also failing. If the execution successed, it sets the output and success.
```
Comment (Sawthis) [inline .github/actions/image-builder/README.md:93]:
```suggestion
If execution failed, the github action fails. If the execution successes, it sets the output and success.
```

File: .github/actions/image-builder/README.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            arg1=value1
+            arg2=value2
+          tags: |
+            latest
+            main
+            commit=${{ .ShortSHA }}
+          export-tags: true
+          image-name: 'ginkgo'
+          dockerfile: 'Dockerfile'
+          env-file: '.env'
```
Comment (Sawthis) [inline .github/actions/image-builder/README.md:115]:
```suggestion
          env-file: 'envs'
```

**Review by Sawthis — CHANGES_REQUESTED** [review]

**Review by Sawthis — DISMISSED** [review]

File: .github/actions/image-builder/README.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+## **build-args**
+
+Optional parameter that allows you to set additional build arguments.
+Each argument has format `name=value`, multiple arguments are separated by new line.
+
+Default: **''**
+
+## **tags**
+
+Optional parameter that allows you to pass tags with which image will be build. Each tag is placed in the new line.
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:48]:
```suggestion
Optional parameter to pass tags with which the image is built. Each tag is placed in a new line.
```

File: .github/actions/image-builder/README.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+Default: **''**
+
+## **export-tags**
+
+Optional parameter that enable exporting tags provided via **tags** input and default tags to be exported as build argument.
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:54]:
```suggestion
Optional parameter that enables exporting tags provided with the **tags** input and default tags to be exported as build argument.
```

File: .github/actions/image-builder/README.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+Each tag will have format **TAG_x**, where `x` is the tag name passed along with the tag.
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:55]:
```suggestion
Each tag gets the format **TAG_x**, where `x` is the tag name passed along with the tag.
```

File: .github/actions/image-builder/README.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+Default: **false**
+
+## **config**
+
+Optional parameter which sets the config file containing information about connection to the Azure DevOps.
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:61]:
```suggestion
Optional parameter that sets the config file containing information about connection to the Azure DevOps.
```

File: .github/actions/image-builder/README.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+Default: **'./configs/kaniko-build-config.yaml'**
+
+## **env-file**
+
+Optional parameter that allows you to provide path to the file with environment variables which loads in the build
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:67]:
```suggestion
Optional parameter to provide a path to the file with environment variables that loads in the build.
```
What is the thing that does the loading - the file, or the environment variables?
If it's the variables, you need plural:

```suggestion
Optional parameter to provide a path to the file with environment variables that load in the build.
```

File: .github/actions/image-builder/README.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+Default: **''**
+
+## **dry-run**
+
+Optional parameter that allows you to prevent calling Azure service
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:73]: We need an article, but I don't understand the context enough to know which
Either
```suggestion
Optional parameter to prevent calling an Azure service
```
Or
```suggestion
Optional parameter to prevent calling the Azure service
```

File: .github/actions/image-builder/README.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+
+Result status of the Azure DevOps execution
+
+## **images**
+
+Formatted JSON array containing all built images.
+
+# How It Works?
+
+Github action is using **europe-docker.pkg.dev/kyma-project/prod/image-builder** docker image to trigger pipeline in Azure DevOps (ADO). It passes parameters using REST API provided by ADO and kaniko-build-config from `main` branch of `test-infra` repository.
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:89]:
```suggestion
The image builder action uses a **europe-docker.pkg.dev/kyma-project/prod/image-builder** Docker image to trigger a pipeline in Azure DevOps (ADO). It passes parameters using REST API provided by ADO and kaniko-build-config from the `main` branch of `test-infra` repository.
```

File: .github/actions/image-builder/README.md [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
+During the ADO pipeline execution this action is checking for status to be reported, if execution ended it fetches status and logs.
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:91]:
```suggestion
During the ADO pipeline execution, the image builder action is checks for the status to be reported. When the execution ends, it fetches the status and logs.
```

File: .github/actions/image-builder/README.md [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+
+If execution failed, the github action fails. If the execution successes, it sets the output and success.
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:93]:
```suggestion
If the execution failed, the the image builder action fails. If the execution succeeds, it sets the output and success status.
```
I'm guessing "success" is a kind of status

Comment (KacperMalachowski) [reply .github/actions/image-builder/README.md:93]: That's correct

File: .github/actions/image-builder/README.md (hunk 10 above)
Comment (KacperMalachowski) [reply .github/actions/image-builder/README.md:67]: It's more like 'that image-builder loads during the build'. We're loading variables from provided file path into azure devops environment.

File: .github/actions/image-builder/README.md [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+
+This action is supposed to trigger image-builder service. It receives authentication tokens (**oidc** and **azure personal access token**) via inputs as well as image name.
+
+Optionally you can set other inputs to control which image with which context should be built.
+
+# Inputs
+
+## **oidc-token**
+
+JWT token exposed by **expose-jwt** github action, which is JWT token signed by github, containing information about running workflow.
```
[the remark this answers is not in the document]
Comment (KacperMalachowski) [reply .github/actions/image-builder/README.md:11]:
```suggestion
A OIDC token exposed by the **expose-jwt** GitHub action, which is a JWT token signed by GitHub, containing information about running workflow.
```

File: .github/actions/image-builder/README.md [hunk 16]
```diff
@@ -0,0 +1,117 @@
+# Image Builder Action
+
+This action is supposed to trigger image-builder service. It receives authentication tokens (**oidc** and **azure personal access token**) via inputs as well as image name.
+
+Optionally you can set other inputs to control which image with which context should be built.
+
+# Inputs
+
+## **oidc-token**
```
[the remark this answers is not in the document]
Comment (KacperMalachowski) [reply .github/actions/image-builder/README.md:9]: I would like to leave it as it follow the expose-jwt style.

File: .github/actions/image-builder/README.md [hunk 17]
```diff
@@ -0,0 +1,117 @@
...
+
+## **config**
+
+Optional parameter that sets the config file containing information about connection to the Azure DevOps.
+
+Default: **'./configs/kaniko-build-config.yaml'**
+
+## **env-file**
+
+Optional parameter to provide a path to the file with environment variables that loads in the build.
```
Comment (NHingerl) [inline .github/actions/image-builder/README.md:67]:
```suggestion
Optional parameter that provides the path to the file with environment variables that are loaded into the build environment.
```

**Review by NHingerl — APPROVED** [review]

File: cmd/image-builder/documentation_assets/image-builder-block-diagram.png
Comment (dekiel) [inline cmd/image-builder/documentation_assets/image-builder-block-diagram.png:1]: In the workflow we are missing component for getting a ADO PAT from secret manager in GCP.
The component "Prepare arguments" is a part of GitHub Action not a workflow. I think it fits rather to the activity diagram not a component diagram. If we would like to specify some how workflow components interact with other components we should present three of them. OIDC getter, ADO PAT getter and Image-builder.

Additionally SAP Vault interaction can be removed. It's not true anymore.

File: .github/actions/image-builder/README.md [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+
+Result status of the Azure DevOps execution
+
+## **images**
+
+Formatted JSON array containing all built images.
+
+# How It Works?
+
+The image builder action uses a **europe-docker.pkg.dev/kyma-project/prod/image-builder** Docker image to trigger a pipeline in Azure DevOps (ADO). It passes parameters using REST API provided by ADO and kaniko-build-config from the `main` branch of `test-infra` repository.
```
Comment (dekiel) [inline .github/actions/image-builder/README.md:89]: My understanding of this sentence is that ADO and kaniko-build-config provide parameters values which are passed to the REST API. This is not true. Could you please rephrase it to make it clear that ADO expose some REST API and we use data from kaniko-build-config to access this API.

File: .github/actions/image-builder/README.md
Comment (dekiel) [inline .github/actions/image-builder/README.md:1]: Please consider replacing the list of inputs and outputs with link to the code and moving the description to the code itself. This way we will not have update list every time we change inputs or outputs. Moreover, the detailed description will exist together with code.

Should we mention that this action requires some environment variables to be present and that it uses these variables values to set values of parameters passed to the ADO pipeline?

File: cmd/image-builder/documentation_assets/image-builder-activity-diagram.png
Comment (dekiel) [inline cmd/image-builder/documentation_assets/image-builder-activity-diagram.png:1]: Please add activity for getting ADO PAT from GCP Secret Manager in a github.com Workflow pane.

**Review by dekiel — CHANGES_REQUESTED** [review]

File: .github/actions/image-builder/README.md (hunk 18 above)
Comment (KacperMalachowski) [reply .github/actions/image-builder/README.md:89]: Sure

File: .github/actions/image-builder/README.md
Comment (KacperMalachowski) [reply .github/actions/image-builder/README.md:1]: The url must be maintained as well, will we remember about that?
I definitely prefer to create this section with some automation. As for now, when the action is used only by our reusable workflow I will make that change.

File: cmd/image-builder/documentation_assets/image-builder-block-diagram.png
Comment (KacperMalachowski) [reply cmd/image-builder/documentation_assets/image-builder-block-diagram.png:1]: Fixed

**Review by dekiel — APPROVED** [review]

File: .github/actions/image-builder/README.md [hunk 19]
```diff
@@ -0,0 +1,43 @@
+# Image Builder Action
+
+The image builder action triggers the image-builder service. It receives authentication tokens (**oidc** and **azure personal access token**) from inputs as well as from the image name.
+
+Optionally, you can set other inputs to control which image with which context should be built.
+
+# Inputs
+Description of each input is available [here](https://github.com/kyma-project/test-infra/blob/main/.github/actions/image-builder/action.yml#L3-L44).
```
Comment (IwonaLanger) [inline .github/actions/image-builder/README.md:8]:
```suggestion
Descriptions of each input is available in the [`action`](https://github.com/kyma-project/test-infra/blob/main/.github/actions/image-builder/action.yml#L3-L44) file.
```

File: .github/actions/image-builder/README.md [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+
+# Outputs
+Description of each output is available [here](https://github.com/kyma-project/test-infra/blob/main/.github/actions/image-builder/action.yml#L46-L52).
```
Comment (IwonaLanger) [inline .github/actions/image-builder/README.md:11]:
```suggestion
Descriptions of each output is available in the [`action`](https://github.com/kyma-project/test-infra/blob/main/.github/actions/image-builder/action.yml#L46-L52) file.
```

File: .github/actions/image-builder/README.md [hunk 21]
```diff
@@ -0,0 +1,43 @@
+# Image Builder Action
+
+The image builder action triggers the image-builder service. It receives authentication tokens (**oidc** and **azure personal access token**) from inputs as well as from the image name.
```
Comment (IwonaLanger) [inline .github/actions/image-builder/README.md:3]:
```suggestion
The image-builder action triggers the image-builder service. It receives authentication tokens (**oidc** and **azure personal access token**) from inputs as well as from the image name.
```

File: .github/actions/image-builder/README.md [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+
+# Inputs
+Description of each input is available [here](https://github.com/kyma-project/test-infra/blob/main/.github/actions/image-builder/action.yml#L3-L44).
+
+# Outputs
+Description of each output is available [here](https://github.com/kyma-project/test-infra/blob/main/.github/actions/image-builder/action.yml#L46-L52).
+
+# How It Works?
+
+The image builder action uses a **europe-docker.pkg.dev/kyma-project/prod/image-builder** Docker image to trigger a pipeline in Azure DevOps (ADO). It passes parameters using REST API provided by ADO. The definition of parameters passed via REST API is taken from kaniko-build-config located on the `main` branch of `test-infra` repository and github context.
```
Comment (IwonaLanger) [inline .github/actions/image-builder/README.md:15]:
```suggestion
The image-builder action uses a **europe-docker.pkg.dev/kyma-project/prod/image-builder** Docker image to trigger a pipeline in Azure DevOps (ADO). It passes parameters using REST API provided by ADO. The definition of parameters passed through REST API is taken from kaniko-build-config located on the `main` branch of the `test-infra` repository and GitHub context.
```

File: .github/actions/image-builder/README.md [hunk 23]
```diff
@@ -0,0 +1,43 @@
...
+
+During the ADO pipeline execution, the image builder action is checks for the status to be reported. When the execution ends, it fetches the status and logs.
```
Comment (IwonaLanger) [inline .github/actions/image-builder/README.md:17]:
```suggestion
During the ADO pipeline execution, the image-builder action checks for the status to be reported. When the execution ends, it fetches the status and logs.
```

File: .github/actions/image-builder/README.md [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+
+If the execution failed, the the image builder action fails. If the execution succeeds, it sets the output and success status.
```
Comment (IwonaLanger) [inline .github/actions/image-builder/README.md:19]:
```suggestion
If the execution fails, the image-builder action fails. If the execution succeeds, it sets the output and success status.
```

**Review by IwonaLanger — APPROVED** [review]

**Review by dekiel — APPROVED** [review]
